sonarqube-companion-rest/src/main/java/pl/consdata/ico/sqcompanion/statistics/StatisticsService.java

```diff
@@ -0,0 +1,62 @@
+package pl.consdata.ico.sqcompanion.statistics;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.consdata.ico.sqcompanion.repository.Project;
+import pl.consdata.ico.sqcompanion.repository.RepositoryService;
+import pl.consdata.ico.sqcompanion.users.metrics.UserStatisticsService;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@Slf4j
+@Service
+public class StatisticsService {
+
+    private final RepositoryService repositoryService;
+    private UserStatisticsService userStatisticsService;
+
+    public StatisticsService(final RepositoryService repositoryService,
+                             final UserStatisticsService userStatisticsService) {
+        this.repositoryService = repositoryService;
+        this.userStatisticsService = userStatisticsService;
+    }
+
+    @Transactional
+    public void syncStatistics() {
+        repositoryService
+                .getRootGroup()
+                .accept(
+                        gr -> gr.getProjects().forEach(project -> this.syncProjectStatsAndCatch(project, gr.getStatistics()))
```

Comment:
Żeby to miało sens w naszym drzewie trzeba by tutaj zrobić getAllProjects i jeszcze się zabezpieczyć żeby nie pobierać jednego projektu kilka razy

Reply:
Fakt, miałem to poprawić zapomniałem ;)